csharp/ql/test/library-tests/dataflow/external-models/steps.ql

```diff
@@ -0,0 +1,41 @@
+import csharp
+import DataFlow
+import semmle.code.csharp.dataflow.ExternalFlow
+import semmle.code.csharp.dataflow.internal.FlowSummaryImpl as FlowSummaryImpl
+import CsvValidation
+
+class SummaryModelTest extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        //"namespace;type;overrides;name;signature;ext;inputspec;outputspec;kind",
+        "My.Qltest;C;false;StepArgRes;(object);;Argument[0];ReturnValue;taint",
+        "My.Qltest;C;false;StepArgArg;(object,object);;Argument[0];Argument[1];taint",
+        "My.Qltest;C;false;StepArgQual;(object);;Argument[0];Argument[-1];taint",
+        "My.Qltest;C;false;StepQualRes;();;Argument[-1];ReturnValue;taint",
+        "My.Qltest;C;false;StepQualArg;(object);;Argument[-1];Argument[0];taint",
+        "My.Qltest;C;false;StepFieldGetter;();;Field[My.Qltest.C.Field] of Argument[-1];ReturnValue;value",
+        "My.Qltest;C;false;StepFieldSetter;(int);;Argument[0];Field[My.Qltest.C.Field] of Argument[-1];value",
+        "My.Qltest;C;false;StepPropertyGetter;();;Property[My.Qltest.C.Property] of Argument[-1];ReturnValue;value",
+        "My.Qltest;C;false;StepPropertySetter;(int);;Argument[0];Property[My.Qltest.C.Property] of Argument[-1];value",
+        "My.Qltest;C;false;StepElementGetter;();;Element of Argument[-1];ReturnValue;value",
+        "My.Qltest;C;false;StepElementSetter;(int);;Argument[0];Element of Argument[-1];value",
+        "My.Qltest.C;Generic<>;false;StepGeneric;(T);;Argument[0];ReturnValue;value",
+        "My.Qltest.C;Generic<>;false;StepGeneric2;(S);;Argument[0];ReturnValue;value"
```

Comment:
I just noticed this: `My.Qltest.C` is actually not a namespace, but a container class. Is this expected? Shouldn't we instead have `My.Qltest` as a namespace, and `C+Generic<>` as the class name?

Reply:
Java seems to be using this notation for nested classes: `Base64$Decoder`. See https://github.com/github/codeql/blob/main/java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll#L256-L263.

Reply:
You are right. I will change this.